src/main/java/org/opensearch/neuralsearch/search/HybridTopScoreDocCollector.java

```diff
@@ -47,20 +47,55 @@ public HybridTopScoreDocCollector(int numHits, HitsThresholdChecker hitsThreshol
     }
 
     @Override
-    public LeafCollector getLeafCollector(LeafReaderContext context) throws IOException {
+    public LeafCollector getLeafCollector(LeafReaderContext context) {
         docBase = context.docBase;
 
-        return new TopScoreDocCollector.ScorerLeafCollector() {
+        return new LeafCollector() {
             HybridQueryScorer compoundQueryScorer;
 
             @Override
             public void setScorer(Scorable scorer) throws IOException {
-                super.setScorer(scorer);
-                compoundQueryScorer = (HybridQueryScorer) scorer;
+                if (scorer instanceof HybridQueryScorer) {
+                    log.debug("passed scorer is of type HybridQueryScorer, saving it for collecting documents and scores");
+                    compoundQueryScorer = (HybridQueryScorer) scorer;
+                } else {
+                    compoundQueryScorer = getHybridQueryScorer(scorer);
+                    if (Objects.isNull(compoundQueryScorer)) {
+                        log.error(
+                            String.format(Locale.ROOT, "cannot find scorer of type HybridQueryScorer in a hierarchy of scorer %s", scorer)
+                        );
+                    }
+                }
+            }
+
+            private HybridQueryScorer getHybridQueryScorer(final Scorable scorer) throws IOException {
```

Comment:
I'm a little worried about this logic, because it lifts the `HybridQueryScorer` up to the top, which doesn't necessarily sound right.

If you have a query like:

```
{
  "bool": {
    "should": [
      {
        "match": { ... }
      },
      {
        "hybrid": { ... }
      }
    ]
}
```

The score should be based on the sum of the match query and the hybrid query scores. From the logic in `collect`, it looks like it will just use the hybrid query score, right?

Reply:
@msfroh this usecase is not supported. Hybrid query clause should be a top level query clause. We have such query we throw exceptions out.

Reply:
[This is plugin code](https://github.com/martin-gaievski/neural-search/blob/hybrid_query_scorers_with_approximations/src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java#L149-L150) that does this validation